pkg/tests/tasks/traffic/ingress/ingress_without_tls_termination_test.go

```diff
@@ -15,69 +15,58 @@
+		t.Log("This test validates configuring an Gateway with TLS PassThrough")
+		t.Log("Doc reference: https://istio.io/v1.14/docs/tasks/traffic-management/ingress/ingress-sni-passthrough/")
 
-	log.Log.Info("Verify NGINX server")
-	pod, _ := util.GetPodName("bookinfo", "run=my-nginx")
-	cmd := fmt.Sprintf(`curl -sS -v -k --resolve nginx.example.com:8443:127.0.0.1 https://nginx.example.com:8443`)
-	msg, err := util.PodExec("bookinfo", pod, "istio-proxy", cmd, true)
-	util.Inspect(err, "failed to get response", "", t)
-	if !strings.Contains(msg, "Welcome to nginx") {
-		t.Errorf("Expected Welcome to nginx; Got unexpected response: %s", msg)
-		log.Log.Errorf("Expected Welcome to nginx; Got unexpected response: %s", msg)
-	} else {
-		log.Log.Infof("Success. Get expected response: %s", msg)
-	}
+		ns := "bookinfo"
 
-	t.Run("TrafficManagement_ingress_configure_ingress_gateway_without_TLS_Termination", func(t *testing.T) {
-		defer util.RecoverPanic(t)
+		t.Cleanup(func() {
+			oc.DeleteFromString(t, ns, nginxIngressGateway)
+			app.Uninstall(t, app.Nginx(ns))
+		})
 
-		log.Log.Info("Configure an ingress gateway")
-		if err := util.KubeApplyContents("bookinfo", nginxIngressGateway); err != nil {
-			t.Errorf("Failed to configure NGINX ingress gateway")
-			log.Log.Errorf("Failed to configure NGINX ingress gateway")
-		}
-		time.Sleep(time.Duration(30) * time.Second)
+		t.LogStep("Create NGINX Deployment")
+		app.InstallAndWaitReady(t, app.Nginx(ns))
 
-		url := "https://nginx.example.com:" + secureIngressPort
-		resp, err := util.CurlWithCA(url, gatewayHTTP, secureIngressPort, "nginx.example.com", nginxServerCACert)
-		defer util.CloseResponseBody(resp)
-		util.Inspect(err, "Failed to get response", "", t)
+		t.LogStep("Verify NGINX server is running by connecting to it via loopback")
+		retry.UntilSuccess(t, func(t test.TestHelper) {
+			oc.Exec(t,
+				pod.MatchingSelector("run=my-nginx", ns),
+				"istio-proxy",
+				"curl -sS -v -k --resolve nginx.example.com:8443:127.0.0.1 https://nginx.example.com:8443",
+				assert.OutputContains(
+					"Welcome to nginx",
+					"Got expected Welcome to nginx response",
+					"Expected to receive response Welcome to nginx, but failed"))
+		})
 
-		bodyByte, err := ioutil.ReadAll(resp.Body)
-		util.Inspect(err, "Failed to read response body", "", t)
+		t.LogStep("Configure Gateway resource with TLS passthrough for host nginx.example.com")
+		oc.ApplyString(t, ns, nginxIngressGateway)
 
-		if strings.Contains(string(bodyByte), "Welcome to nginx") {
-			log.Log.Info(string(bodyByte))
-		} else {
-			t.Errorf("Failed to get Welcome to nginx: %v", string(bodyByte))
-		}
+		t.LogStep("Verify NGINX is reachable from outside the cluster through the ingressgateway")
+		gatewayHTTP := istio.GetIngressGatewayHost(t, meshNamespace)
+		secureIngressPort := istio.GetIngressGatewaySecurePort(t, meshNamespace)
+		retry.UntilSuccess(t, func(t test.TestHelper) {
+			curl.Request(t,
```

Comment:
@luksa  This curl request is failing in my OCP test.
I am getting curl failed to resolve error. Did the test pass from your side ?

> curl -v --resolve "nginx.example.com:$SECURE_INGRESS_PORT:$INGRESS_HOST" --cacert $CERT > "https://nginx.example.com:$SECURE_INGRESS_PORT"
> * Couldn't parse CURLOPT_RESOLVE entry 'nginx.example.com:443:istio-ingressgateway-istio-system.apps.yuaxu-maistra-daily.devcluster.openshift.com'!

Reply:
Yes, it works. But the error message you pasted here doesn't have sense. We're not even calling curl here, but making a request via Go's http package, so how can the error come from curl?

Reply:
I was troubleshooting it with a manual curl command.
Nvm, it must be my local network or DNS issue. I run it again and this PR test is Pass from my side now.
Thanks